### PR TITLE
Move more of the controller out of hack and into v2

### PR DESCRIPTION
**Pull request by matthchr**
There are other problems we should solve (the crossplane directory and generator), but this at least removes the circular reference we had which was causing me a lot of issues
**How does this PR make you feel**:
gif

**If applicable**:
- [x] this PR contains tests

### CODE REVIEW

**Comment by matthchr** [comment]
Hmm I think I broke the Docker build for the controller image - fixing that...

**Comment by matthchr** [comment]
Will be waiting for @porges approval on this (to ensure I didn't screw it up and didn't collide with anything he's doing)

**Comment by matthchr** [comment]
Fixed the Dockerfile stuff

File: hack/generator/pkg/config/configuration.go
Comment (Porges) [inline hack/generator/pkg/config/configuration.go:65]: this was added but not actually used in the config file?

File: hack/generator/azure-arm.yaml
Comment (Porges) [inline hack/generator/azure-arm.yaml:6]: one change I made when sketching this was to have the `typesOutputPath` be relative to the destination module file (since `typeRegistrationOutputFile` already is), but I can make this change later

**Review by Porges — APPROVED** [review]

File: hack/generator/pkg/config/configuration.go
Comment (matthchr) [reply hack/generator/pkg/config/configuration.go:65]: It's not for the config file - it's cached and set during `initialize` (mostly to avoid having to return err or panic where I need it)

**Pull request merged**
